ngraph/core/src/op/non_max_suppression.cpp

```diff
@@ -641,36 +624,109 @@ op::v5::NonMaxSuppression::NonMaxSuppression(
+                                                           m_output_type);
+        break;
+    case 5:
+        return std::make_shared<op::v5::NonMaxSuppression>(new_args.at(0),
+                                                           new_args.at(1),
+                                                           new_args.at(2),
+                                                           new_args.at(3),
+                                                           new_args.at(4),
+                                                           m_box_encoding,
+                                                           m_sort_result_descending,
+                                                           m_output_type);
+        break;
+    default:
+        return std::make_shared<op::v5::NonMaxSuppression>(new_args.at(0),
+                                                           new_args.at(1),
+                                                           new_args.at(2),
+                                                           new_args.at(3),
+                                                           new_args.at(4),
+                                                           new_args.at(5),
+                                                           m_box_encoding,
+                                                           m_sort_result_descending,
+                                                           m_output_type);
+        break;
+    }
+}
 
-    return std::make_shared<op::v5::NonMaxSuppression>(new_args.at(0),
-                                                       new_args.at(1),
-                                                       arg2,
-                                                       arg3,
-                                                       arg4,
-                                                       arg5,
-                                                       m_box_encoding,
-                                                       m_sort_result_descending,
-                                                       m_output_type);
+using V5BoxEncoding = op::v5::NonMaxSuppression::BoxEncodingType;
+
+namespace
+{
+    constexpr size_t boxes_port = 0;
+    constexpr size_t scores_port = 1;
+    constexpr size_t max_output_boxes_port = 2;
+    constexpr size_t iou_threshold_port = 3;
+    constexpr size_t score_threshold_port = 4;
+    constexpr size_t soft_nms_sigma_port = 5;
+
+    inline bool is_float_type_admissible(const element::Type& t)
+    {
+        return t == element::f32 || t == element::f16 || t == element::bf16;
+    }
+
+    inline bool is_scalar_or_1d_tensor_with_1_element(const PartialShape& p)
+    {
+        if (p.is_dynamic())
+        {
+            return false;
+        }
+
+        Shape shape = p.to_shape();
+
+        return is_scalar(shape) || (is_vector(shape) && (shape[0] == 1));
+    }
+
+    float float_from_constant_node(const std::shared_ptr<op::Constant>& constant,
```

Comment:
The `cast_vector<float>` does exactly the same. Please, consider removing this function.

Reply:
Done.